ContributedExamples/surface_water-mass_transformation.ipynb

```diff
@@ -0,0 +1,782 @@
+    "    and freshwater) referenced to 1000 db from monthly ACCESS-OM2 output.\n",
+    "    Suitable for analysis of high-resolution (0.1 degree) output (the scattered .load()'s allowed this)\n",
+    "    \n",
+    "    expt - text string indicating the name of the experiment\n",
+    "    session - a database session created by cc.database.create_session()  \n",
+    "    start_time - text string designating the start month of the analysis ('YYYY-MM', e.g. '1905-01')\n",
+    "    end_time - text string indicating the end month of the analysis ('YYY-MM', e.g. '1905-12')\n",
+    "    outpath - text string indicating directory where output databases are to be saved (3 xarray databases, can\n",
+    "    modify to combine these if memory permits)\n",
+    "    lat_north - function computed processes between lat = -90 and lat = lat_north\n",
+    "    n - designate if a subset of output files is to be considered (see cc.querying.getvar)\n",
+    "    \n",
+    "    NOTE: this function assumes you are averaging over an integer number of years (though the start month \n",
+    "    need not be january, e.g. can have start_time = '1905-05', end_time = '1907-04' etc), modify if otherwise.\n",
+    "    \n",
+    "    NOTE: assumes surface_temp and surface_salt variables are in potential temperature (K) and practical\n",
+    "    salinity (PSU), simplifications may be made if conservative temperature (C) and absolute salinity (g/kg)\n",
+    "    are computed online\n",
+    "    \n",
+    "    required modules:\n",
+    "    xarray as xr\n",
+    "    numpy as np\n",
+    "    cosima_cookbook as cc\n",
+    "    from gsw import alpha, SA_from_SP, p_from_z, CT_from_pt, beta, sigma1 \n",
+    "    '''\n",
+    "    ## getvar all required variables\n",
+    "    SST = cc.querying.getvar(expt,'surface_temp',session) - 273.15 # SST - conservative temperature in K (sheck this is the case for your run)\n",
+    "    SSS_PSU = cc.querying.getvar(expt,'surface_salt',session) # SSS - practical salinity (not absolute)\n",
+    "    pme_river = cc.querying.getvar(expt,'pme_river',session) # mass flux of precip - evap + river\n",
+    "    ## getvar the components of the net surface heat fux instead of the net_surface_heating variable\n",
+    "    sfc_hflux_from_runoff = cc.querying.getvar(expt,'sfc_hflux_from_runoff',session) # W/m2\n",
+    "    sfc_hflux_coupler = cc.querying.getvar(expt,'sfc_hflux_coupler',session) # W/m2\n",
+    "    sfc_hflux_pme = cc.querying.getvar(expt,'sfc_hflux_pme',session) # W/m2\n",
+    "    frazil_3d_int_z = cc.querying.getvar(expt,'frazil_3d_int_z',session) # W/m2\n",
+    "    geolon_t = cc.querying.getvar(expt,'geolon_t',session, n=1)\n",
+    "    geolat_t = cc.querying.getvar(expt,'geolat_t',session, n=1)\n",
+    "    ## slice for time and latitudinal constraints\n",
+    "    SST = SST.sel(time = slice(start_time,end_time)).sel(yt_ocean = slice(-90,lat_north))\n",
```

Comment:
You can define your time/space slices as variables at the top, and merge the selections, e.g.

```python
time_slice = slice(start_time, end_time)
lat_slice = slice(-90, lat_north)
...
var = var.sel(time=time_slice, yt_ocean=lat_slice)
```